server/pulp/server/async/scheduler.py

```diff
@@ -121,30 +113,39 @@ def check_celery_processes(self):
         present. If there are zero of either, log at the error level that Pulp will not operate
         correctly.
         """
-        msg = _('Checking if pulp_workers, pulp_celerybeat, or pulp_resource_manager '
-                'processes are missing for more than %d seconds') % self.CELERY_TIMEOUT_SECONDS
+        msg = _('Checking if pulp_workers, pulp_celerybeat, or pulp_resource_manager processes '
+                'are missing for more than %d seconds') % constants.CELERY_TIMEOUT_SECONDS
         _logger.debug(msg)
         now = ensure_tz(datetime.utcnow())
-        oldest_heartbeat_time = now - timedelta(seconds=self.CELERY_TIMEOUT_SECONDS)
+        oldest_heartbeat_time = now - timedelta(seconds=constants.CELERY_TIMEOUT_SECONDS)
         worker_list = Worker.objects.all()
         worker_count = 0
         resource_manager_count = 0
         scheduler_count = 0
+
         for worker in worker_list:
             if worker.last_heartbeat < oldest_heartbeat_time:
                 msg = _("Worker '%s' has gone missing, removing from list of workers") % worker.name
                 _logger.error(msg)
-                _delete_worker(worker.name)
-            elif worker.name.startswith(SCHEDULER_WORKER_NAME):
+
+                # Check if worker is a scheduler, and clean up appropriately
```

Comment:
As you mentioned on IRC, `delete()` is mongo defined so let's leave that one alone. [_delete_worker](https://github.com/dralley/pulp/blob/72b05d94ad312821c6ddd04805c79ae429774aea/server/pulp/server/async/tasks.py#L227) is something we define though.